core/src/commonMain/kotlin/com/powersync/attachments/implementation/AttachmentContextImpl.kt

```diff
@@ -0,0 +1,194 @@
+package com.powersync.attachments.implementation
+
+import co.touchlab.kermit.Logger
+import com.powersync.PowerSyncDatabase
+import com.powersync.attachments.Attachment
+import com.powersync.attachments.AttachmentContext
+import com.powersync.attachments.AttachmentState
+import com.powersync.db.getString
+import com.powersync.db.internal.ConnectionContext
+import kotlinx.datetime.Clock
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+
+public open class AttachmentContextImpl(
```

Comment:
Same here, wondering if this can be `internal`?

Reply:
Other SDKs usually allow overriding the AttachmentQueue implementation. I'd like to keep these defaults public in order for devs to extend them if necessary. I've added a comment to these classes.